app/assets/javascripts/discourse/app/app.js

```diff
@@ -55,10 +55,43 @@ class Discourse extends Application {
     _pluginCallbacks.push({ version, code });
   }
 
+  _cssReady() {
+    const links = document.getElementsByClassName("async-css-loading");
+
+    if (links.length === 0) {
+      return true;
+    }
+
+    return [...links].every((link) => {
+      return link.dataset.processed === "true";
+    });
+  }
+
+  _displayApp(callback) {
+    document.dispatchEvent(new CustomEvent("discourse-ready"));
+    document.getElementById("main").removeAttribute("style");
```

Comment:
This could potentially undo styles which are introduced by a theme. So I think we should be more surgical, and only remove the `display:none`

Reply:
Good point. I decided to go with an inline style element and remove a class instead.